### PR TITLE
Tokenizer refactor

### PR DESCRIPTION
**Pull request by znichola**
Changing the way the make file works and internal structure of the unit test.
Should be neater and clearer on how to adapt it for later.

### CODE REVIEW

File: srcs/tokenizer.c [hunk 1]
```diff
@@ -52,14 +59,31 @@ int	check_token_literals(char *str)
 	return (-1);
 }
 
+static void	string_decetion(t_token *tok, char **str)
```
Comment (t-h2o) [inline srcs/tokenizer.c:62]: ???

Comment (znichola) [reply srcs/tokenizer.c:62]: yes, well sometimes I make typs not but not oftne

File: unit-tests/run_test_tokenizer.sh [hunk 2]
```diff
@@ -18,11 +18,11 @@ function exec_test()
 	# local DIFF_RES=$(diff <(echo "$2") <(echo "$RES"))
 	# cat -e <(echo $DIFF_RES)
 	if [[ $RES == $3 ]]; then
-		echo "${BLUE}$1 ${BOLD}${GREEN}OK${RESET}"
+		printf "${BLUE}%2s ${BOLD}${GREEN}OK${RESET}\n" "$1"
```
Comment (t-h2o) [inline unit-tests/run_test_tokenizer.sh:22]: Better `printf` than `echo` 👍

Comment (znichola) [reply unit-tests/run_test_tokenizer.sh:22]: ok so it's about potability? will look into it.

File: unit-tests/tokenizer_test.c [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
 /*                                                        :::      ::::::::   */
-/*   test_tokenizer.c                                   :+:      :+:    :+:   */
+/*   tokenizer_test.c                                   :+:      :+:    :+:   */
 /*                                                    +:+ +:+         +:+     */
 /*   By: znichola <znichola@student.42lausanne.ch>  +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2023/01/11 19:01:30 by znichola          #+#    #+#             */
-/*   Updated: 2023/01/12 19:54:10 by znichola         ###   ########.fr       */
+/*   Updated: 2023/01/12 20:28:06 by znichola         ###   ########.fr       */
 /*                                                                            */
```
Comment (t-h2o) [inline unit-tests/tokenizer_test.c:10]: Why did you change the header but not the file content?
